lms/djangoapps/courseware/masquerade.py

```diff
@@ -19,16 +25,21 @@
 # The value is a dict from course keys to CourseMasquerade objects.
 MASQUERADE_SETTINGS_KEY = 'masquerade_settings'
 
+# The key used to store temporary XBlock field data in the Django session.  This is where field
+# data is stored to avoid modifying the state of the user we are masquerading as.
+MASQUERADE_DATA_KEY = 'masquerade_data'
+
 
 class CourseMasquerade(object):
     """
     Masquerade settings for a particular course.
     """
-    def __init__(self, course_key, role='student', user_partition_id=None, group_id=None):
+    def __init__(self, course_key, role='student', user_partition_id=None, group_id=None, user_name=None):
         self.course_key = course_key
         self.role = role
         self.user_partition_id = user_partition_id
         self.group_id = group_id
+        self.user_name = user_name
```

Comment:
If you use the "Preview as specific student" functionality, the user you enter is automatically treated as a student, regardless of whether the user you entered actually has staff privileges.  I thought about disallowing to masquerade as a different staff user, but I think the current behaviour is good enough.  (Simply setting role to "staff" here when the username entered is a staff user does not work – it would require adjustment in many places in the code.)